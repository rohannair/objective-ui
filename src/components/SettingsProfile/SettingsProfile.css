@import '../../styles/globals';

.SettingsProfile {
  display: flex;
  flex-direction: row;
  flex-wrap: wrap;
}

.heading {
  margin-bottom: 20px;
  width: 100%;
}

.buttonContainer {
  display: flex;
  width: 100%;
  margin-top: 20px;
  justify-content: flex-end;
}

.form,
.imageForm {
  display: flex;
  flex-direction: column;
}

.imageContainer {
  display: flex;
  width: 250px;
  margin-bottom: 10px;
  border-radius: 3px;
  height: 250px;
  overflow: hidden;
  background-color: $background;
}

.image {
  display: block;
  width: 100%;
  height: auto;
}

.placeholder {
  display: flex;
  width: 100%;
  height: 100%;
  justify-content: center;
  align-items: center;
  font-size: 5rem;

  i {
    color: rgba(0,0,0, 0.1);
  }

}

.form {
  width: calc(100% - 280px);
  margin-left: auto;
}

.avatarButtonContainer {
  display: flex;
  margin-bottom: 10px;
  justify-content: center;

  button {
    height: 1.875rem;
    border-radius: 1.875em;
    font-size: 0.75em !important;
    line-height: 1;
  }
}
