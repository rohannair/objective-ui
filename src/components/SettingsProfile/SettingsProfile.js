import React, { Component, PropTypes } from 'react'
import styles from './SettingsProfile.css'
import classNames from 'classnames/bind'

import Button from '../Button'
import Input from '../Forms/TextInput'
import Uploader from '../Uploader'

let cx = classNames.bind(styles)

class SettingsProfile extends Component {
  constructor(props) {
    super(props)

    const { img, firstName, lastName, jobTitle } = this.props.user

    this.state = {
      img,
      firstName,
      lastName,
      jobTitle
    }
  };

  render() {
    const classname = cx({
      [styles.SettingsProfile]: true
    })

    const user = this.state
    const imgSrc = /^http/.test(user.img)
    ? user.img
    : `data:image/jpeg;base64,${user.img}`

    const img = user.img
    ? <img className={styles.image} src={imgSrc} />
    : <div className={styles.placeholder}><i className="zmdi zmdi-account" /></div>

    return (
      <div className={classname}>
        <h2 className={styles.heading}>Edit Profile</h2>

        <section className={styles.imageForm}>
          <div className={styles.imageContainer}>
            { img }
          </div>
          <div className={styles.avatarButtonContainer}>
            <Uploader
              submitImage={this._handleChange.bind(this, 'img')}
              imageExists={!!this.state.img}
              resizeOptions={{ height: 250, width: 250, type: 'crop' }}>
              <Button primary>Upload New Avatar</Button>
            </Uploader>

          </div>
          <div className={styles.avatarButtonContainer}>
            <Button link onClick={e => {
              e.preventDefault()
              this.setState({
                ...this.state,
                img: ''
              })
            }}>Delete</Button>
          </div>
        </section>

        <section className={styles.form}>
          <Input
            type='text'
            label='First Name'
            value={user.firstName}
            onChange={this._handleChange.bind(this, 'firstName')}
          />

          <Input
            type='text'
            label='Last Name'
            value={user.lastName}
            onChange={this._handleChange.bind(this, 'lastName')}
          />

          <Input
            type='text'
            label='Title'
            value={user.jobTitle}
            onChange={this._handleChange.bind(this, 'jobTitle')}
          />
        </section>

        <section className={styles.buttonContainer}>
          <Button primary onClick={this._handleSubmit}>Confirm</Button>
        </section>
      </div>
    )
  };

  _handleChange = (name, value) => {
    this.setState({
      ...this.state,
      [name]: value
    })
  };

  _handleSubmit = (e) => {
    e.preventDefault()
    this.props.handleSubmit(this.state)
  };

};

export default SettingsProfile
