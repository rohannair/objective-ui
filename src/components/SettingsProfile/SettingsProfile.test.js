import React from 'react'
import SettingsProfile from './index.js'

describe('Test suite for SettingsProfile component', () => {
  it('SettingsProfile should exist', () => {
    let wrapper = shallow(<SettingsProfile/>)
    expect(wrapper).toMatchSnapshot()
  })
})

