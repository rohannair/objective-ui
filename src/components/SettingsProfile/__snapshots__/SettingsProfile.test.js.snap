exports[`Test suite for SettingsProfile component SettingsProfile should exist 1`] = `
<div
  className="undefined">
  <h2>
    Edit Profile
  </h2>
  <section>
    <div>
      <div>
        <i
          className="zmdi zmdi-account" />
      </div>
    </div>
    <div>
      <Button
        onClick={[Function]}
        primary={true}>
        Upload New Avatar
      </Button>
    </div>
    <div>
      <Button
        link={true}
        onClick={[Function]}>
        Delete
      </Button>
    </div>
  </section>
  <section>
    <TextInput
      inputType="text"
      label="First Name"
      onChange={[Function]}
      type="text"
      value="" />
    <TextInput
      inputType="text"
      label="Last Name"
      onChange={[Function]}
      type="text"
      value="" />
    <TextInput
      inputType="text"
      label="Title"
      onChange={[Function]}
      type="text"
      value="" />
  </section>
  <section>
    <Button
      onClick={[Function]}
      primary={true}>
      Confirm
    </Button>
  </section>
</div>
`;
